pyproject.toml

```diff
@@ -0,0 +1,73 @@
+[build-system]
+requires = ["setuptools>=61.0"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "dnslib"
+description = "Simple library to encode/decode DNS wire-format packets"
+authors  = [
+    {name = "PaulC"},
+]
+
+# Dependency Information
+requires-python = ">=3.7"
+dependencies = [
+    "typing-extensions;python_version<'3.11'",
+]
+
+# Extra information
+readme = "README.md"
+license = {text = "BSD"}
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "License :: OSI Approved :: BSD License",
+    "Intended Audience :: Developers",
+    "Topic :: Internet :: Name Service (DNS)",
+    "Typing :: Typed",
+]
+
+dynamic = ["version"]
+
+[project.urls]
+# Homepage = "https://paulc.github.io/dnslib/latest/"
+Homepage = "https://github.com/paulc/dnslib"
+
+[project.optional-dependencies]
+dev = [
+    ## Type Stubs
+    ### dev.sh dependencies
+    ## Formatting / Linting
+    "validate-pyproject[all]",
+    "black",
+    "pylint",
+    "mypy",
+    ## Testing
+    "pytest",
+    ## REPL
+    "bpython",
+]
```

Comment:
TODO: remove these as these are from my own package template (that I've not implemented properly here)